File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr1, vector<vector<int>> arr2) {
    vector<vector<int>> answer;

    int row = arr1.size(), col = arr2[0].size();
    for (int i = 0; i < row; i++) {
        vector<int> line;
        for (int j = 0; j < col; j++) {
            int value = 0;
            for (int k = 0; k < arr2.size(); k++) {
                value += arr1[i][k] * arr2[k][j];
            }
            line.emplace_back(value);
        }
        answer.emplace_back(line);
    }

    return answer;
}

File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.swift
import Foundation

func solution(_ arr1:[[Int]], _ arr2:[[Int]]) -> [[Int]] {
    let (n, m, k) = (arr1.count, arr2[0].count, arr2.count)
    var answer = [[Int]](repeating: [Int](repeating: 0, count: m), count: n)
    
    for i in 0..<n {
        for j in 0..<m {
            //answer[i][j]
            for a in 0..<k {
                answer[i][j] += arr1[i][a] * arr2[a][j]
            }
        }
    }
    
    return answer
}